### PR TITLE
fix(emotion,shared-types): better TS types for theme objects and their overrides

### PR DESCRIPTION
**Pull request by matyasf**
Also convert some of the final files in the docs app to TS and improve their typing

TEST PLAN:
Try to make various theme and override objects, they should have nice autocomplete and no errors

### CODE REVIEW

File: packages/emotion/src/EmotionTypes.ts [hunk 1]
```diff
@@ -87,3 +141,62 @@ export type {
   GenerateStyle,
   ComponentStyle
 }
+/*
```
Comment (matyasf) [inline packages/emotion/src/EmotionTypes.ts:144]: I've left this here intentionally to test this complex type

File: packages/emotion/src/EmotionTypes.ts [hunk 2]
```diff
@@ -29,21 +29,75 @@ import type {
   DeepPartial
```
Comment (matyasf) [inline packages/emotion/src/EmotionTypes.ts:29]: The type changes in this file and the main change

File: packages/__docs__/components.ts [hunk 3]
```diff
@@ -130,4 +130,4 @@ export { Drilldown } from '@instructure/ui-drilldown'
 export { SourceCodeEditor } from '@instructure/ui-source-code-editor'
 export { TopNavBar } from '@instructure/ui-top-nav-bar'
 export { TruncateList } from '@instructure/ui-truncate-list'
-export { canvas, canvasHighContrast, instructure } from '@instructure/ui-themes'
```
Comment (matyasf) [inline packages/__docs__/components.ts:133]: This was left here from v9... we really need to type check the docs app

File: packages/__docs__/src/compileMarkdown.tsx [hunk 4]
```diff
@@ -39,7 +39,32 @@ import { Playground } from './Playground'
...
+function trimIndent(str: string) {
+  const lines = `${str.replace(/\r\n/g, '\n').replace(/\r/g, '\n')}\n`.split(
+    '\n'
+  )
+  let indentFound = false
+  let trimmed = ''
+  let indent = ''
+  lines.forEach((line, _i) => {
+    // matches whitespace at the end of a string
+    line.replace(/\s*$/, '')
```
Comment (ToMESSKa) [inline packages/__docs__/src/compileMarkdown.tsx:52]: I think .replace() does not change the original string, so the whitespace isn't actually removed. The result should be reassigned e.g.: line = line.replace(/\s*$/, '')

**Review by ToMESSKa — CHANGES_REQUESTED** [review]

File: packages/__docs__/src/compileMarkdown.tsx (hunk 4 above)
Comment (matyasf) [reply packages/__docs__/src/compileMarkdown.tsx]: good catch, fixed

**Review by HerrTopi — APPROVED** [review]

**Review by ToMESSKa — APPROVED** [review]

**Pull request merged**